### PR TITLE
Rename UCP names and ContextType with TAE(Tanzu Application Engine)

### LINKED ISSUE #515: Rename the UCP instance names with TAE(Tanzu Application Engine)
**Describe the feature request**
Rename the UCP instance names with TAE(Tanzu Application Engine)

**Describe alternative(s) you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR renames all the instances of UCP names and context Type with TAE(Tanzu Application Engine)resource. Also, perform syncing the kubeconfig context while creating the CLI context for TAE and K8S-Cluster-Endpoint(e.g: TKG with pinniped)

Changes Summary:
- Rename all the instances of UCP names and context Type with TAE(Tanzu Application Engine)resource.
- Sync the kubeconfig and update the current kube context while creating context for TAE and K8s Cluster endpoint(TKG with pinniped) 

### Which issue(s) this PR fixes

Fixes #515 #516 

### Describe testing done for PR
All unit tests and CI passed.

Tested creating the TAE context and validated the currentContext in the kubeconfig file is set to kubecontext created as a result of creating the CLI context for TAE(Tanzu Application Engine)

### Release note

```release-note
Rename UCP names and ContextType with TAE(Tanzu Application Engine)
```

### CODE REVIEW

File: pkg/command/context.go [hunk 1]
```diff
@@ -976,8 +976,9 @@ func listDeactivatedPlugins(deactivatedPlugins []discovery.Discovered, ctxName s
 }
 
 func displayContextListOutputListView(cfg *configtypes.ClientConfig, writer io.Writer) {
+	fmt.Printf("Prem targetStr is %v\n", targetStr)
```
Comment (marckhouzam) [inline pkg/command/context.go:979]: To remove 😄

File: pkg/command/context.go [hunk 2]
```diff
@@ -976,8 +976,9 @@ func listDeactivatedPlugins(deactivatedPlugins []discovery.Discovered, ctxName s
...
 	target := getTarget()
-
+	fmt.Printf("Prem target is %v\n", target)
```
Comment (marckhouzam) [inline pkg/command/context.go:981]: To remove

**Review by marckhouzam — APPROVED** [review]
Two debug printouts to remove (and the go.mod), but after that LGTM

File: pkg/command/context.go (hunk 1 above)
Comment (prkalle) [reply pkg/command/context.go]: Thank you for catching this 😄 . Removed the debug print

File: pkg/command/context.go [hunk 3]
```diff
@@ -492,10 +492,17 @@ func globalUCPLogin(c *configtypes.Context) error {
 	if err != nil {
 		return err
 	}
+	// update the current context in the kubeconfig file after creating the context
```
Comment (prkalle) [inline pkg/command/context.go:495]: @vuil Added logic to update the current kubecontext in the kubeconfig to the kube context created as result of creating CLI context for Kubernetes-cluster-endpoint(TKG with pinniped)

File: pkg/command/context.go [hunk 4]
```diff
@@ -637,6 +644,14 @@ func k8sLogin(c *configtypes.Context) error {
 		if err != nil {
 			return err
 		}
+		// update the current context in the kubeconfig file after creating the context
+		if skipSync, _ := strconv.ParseBool(os.Getenv(constants.SkipUpdateKubeconfigOnContextUse)); !skipSync {
```
Comment (prkalle) [inline pkg/command/context.go:648]: Same as above comment

File: pkg/command/context.go (hunk 3 above)
Comment (vuil) [reply pkg/command/context.go:495]: @prkalle nit(preexisting): probably better to put the skipsync check in syncCurrentKubeContext

Comment (prkalle) [reply pkg/command/context.go:495]: Agreed. Moved skipsync check to syncCurrentKubeContext() method.

**Review by marckhouzam — APPROVED** [review]
LGTM

**Review by vuil — APPROVED** [review]
just a nit, not blocking.
lgtm, thanks!

**Pull request merged**